reana_job_controller/htcondorcern_job_manager.py

```diff
@@ -129,7 +137,22 @@ def execute(self):
         job_ad["TransferInput"] = self._get_input_files()
         job_ad["TransferOutput"] = "."
         job_ad["PeriodicRelease"] = classad.ExprTree("(HoldReasonCode == 35)")
-        job_ad["MaxRunTime"] = 3600
+        if self.htcondor_max_runtime in (
+            "espresso",
+            "microcentury",
+            "longlunch",
+            "workday",
+            "tomorrow",
+            "testmatch",
+            "nextweek",
+        ):
+            job_ad["JobFlavour"] = self.htcondor_max_runtime
+        elif HTCondorJobManagerCERN._is_int(self.htcondor_max_runtime):
```

Comment:
Would it be possible to do this check with something like `isinstance(self.htcondor_max_runtime, int)`? So we avoid that extra function.

Reply:
I agree the extra function excessive. `isinstance(object, type)` seems to return True if the object is the specified type. `self.htcondor_max_runtime` is a string (or NoneType if not set), which causes trouble comparing it to int. I could however take the try-except out of the function and put it here!

Another option to consider is to include a third field `htcondor_job_flavour`. HTC submission files actually uses `+JobFlavour` when specifying flavours like espresso, microcentury etc and `+MaxRuntime` when specifying the runtime in seconds. I decided to combine them into one field since all that flavours do, is bucket the job into a max runtime. At the time it seemed unnecessary to create two fields for what is essentially the same thing. It seems like most people uses flavours when creating jobs so I didn't want to leave it out, however they are quite limiting so allowing seconds is also important. If I separated the two then this wouldn't be a problem since `self.htcondor_max_runtime` would always be int or None.

Reply:
Ah ok! I understand the reasoning now. I found this [`str.isdigit()`](https://docs.python.org/3/library/stdtypes.html#str.isdigit) function that can come in handy in this case, let me know.

As you've explained it, it also make sense to me to use the same field for both options :)

Reply:
Oh yeah `str.isdigit()` exist! I will use that instead :)